src/GraphQL/Builders/FieldBuilder.cs

```diff
@@ -488,4 +489,47 @@ public virtual FieldBuilder<TSourceType, TReturnType> ComplexityImpact(double im
     [AllowedOn<IObjectGraphType>]
     public virtual FieldBuilder<TSourceType, TReturnType> DependsOn<TService>()
         => this.DependsOn(typeof(TService));
+
+    /// <summary>
+    /// Applies middleware to the field. If middleware is already set, the new middleware will be chained after the existing middleware.
+    /// </summary>
+    /// <param name="middleware">The middleware to apply.</param>
+    [AllowedOn<IObjectGraphType>]
+    public virtual FieldBuilder<TSourceType, TReturnType> ApplyMiddleware(IFieldMiddleware middleware)
```

Comment:
Is it possible to add another overload?
```c#
public virtual FieldBuilder<TSourceType, TReturnType> ApplyMiddleware<TFieldMiddleware >()
    where TFieldMiddleware : IFieldMiddleware
```

And resolve it from DI?

Reply:
> And also one that accepts a delegate from simple middlewares

???  I guess I'm not exactly sure what you mean here.

Reply:
done

Reply:
The problem is that it would require a bunch of variations of FieldBuilder like `FieldBuilder<TSource, TReturn, TService1, TService2, TService3>`.  Additionally, the resolver builders are in the MicrosoftDI package, allowing for the scoped option, separate from the field builders in the main package...

So for now we start the resolver builders with `Resolve()`, then chain together the options and stuff, ending the resolver builders with `ResolveAsync(...)` or `Resolve(...)` which brings you back to the base field builder.

We can look at changing this for v9 if you like, but I don't have a suggestion how to do so.